### ISSUE TITLE
Proposal for "openssl_encrypt"

### ISSUE THREAD

**Issue by galaxiesFarApart** [php/doc-en#3761]
Proposal for "openssl_encrypt"
---------------------------------------------
Change the initial function description:
"Encrypts given data with given method and key, returns a raw or base64 encoded string"
     to
"Encrypts given data with given method and passphrase, returns a raw or base64 encoded string"

================================================

Proposal for "openssl_decrypt"
---------------------------------------------
Change the initial function description:
"Takes a raw or base64 encoded string and decrypts it using a given method and key."
      to
"Takes a raw or base64 encoded string and decrypts it using a given method and passphrase."

Change  the Parameter definition of "passphrase":
"The key."
      to
"The passphrase. If the passphrase is shorter than expected, it is silently padded with NUL characters; if the passphrase is longer than expected, it is silently truncated."

_Originally posted by @galaxiesFarApart in https://github.com/php/doc-en/issues/3738#issuecomment-2351541296_

**Comment by damianwadley** [comment]
Every definition of "passphrase" I can see talks about it in terms of human language words that function like a password;

* [a sequence of words or other text... similar to a password in usage](https://en.wikipedia.org/wiki/Passphrase)
* [A passphrase is a memorized secret consisting of a sequence of words or other text](https://csrc.nist.gov/glossary/term/passphrase)
* [a sequence of words, typically longer than a traditional password... usually consists of a combination of words that are easy to remember](https://www.malwarebytes.com/cybersecurity/basics/what-is-passphrase)

In other words, no, it does not make sense to use the word "passphrase" in this context of things that are very much meant to be cryptographically secure.

What about "key"? Consider the following:
![image](https://github.com/user-attachments/assets/e39dfaa5-6511-443c-93e4-e58da685470d)

So "key" is the correct term and the docs should not use "passphrase".

**Comment by cmb69** [comment]
Then I'd go with "key" instead of "passphrase" for *symmetric* encryption. It seems to me that for *asymetric* encryption, the usage of "passphrase" is appropriate when referring to the passphrase which is used to unlock an SSH key, for example. So from quickly skimming openssl.stub.php, the only wrongly named parameters are those for `openssl_encrypt()`  and `openssl_decrypt()`. However, since we cannot change the parameter name for BC reasons, I suggest to clarify this in the parameter description.

**Comment by bukka** [comment]
Except it's not key either because it's getting modified currently (padded with zeros or truncated). Although it's not passphrase either really but the name is there and changing that would be a BC break (named param) so I think we need to stick with passphrase. I added a note about KDF not being used and created a PR: https://github.com/php/doc-en/pull/3774

**Comment by JayEn84** [comment]
I found this issue after I just saw on https://www.php.net/manual/en/function.openssl-encrypt.php that the changed documentation of `$passphrase` is now also the documentation of `$iv`. I think that `$iv` is something else and that its documentation was changed by mistake, right?

![grafik](https://github.com/user-attachments/assets/64bf2087-ecd6-4611-8c5e-68cec994d4b6)

**Comment by cmb69** [comment]
@JayEn84, should be fixed with #3887.

**Comment by bukka** [comment]
This should be all address so closing. Discussion about changing the actual name is php-src thing and there is no agreement so it would likely require RFC.
